Client/Frontend/Home/TopSites/DataManagement/TopSitesManager.swift

```diff
@@ -41,7 +41,10 @@ class TopSitesManager: FeatureFlaggable, HasNimbusSponsoredTiles {
     }
 
     var hasData: Bool {
-        return !historySites.isEmpty
+        // We cannot simply look if there's contiles data to look for other data since it won't necessarely show
+        // depending on the different contiles rules
+        let hasOtherData = shouldAddSponsoredTiles || googleTopSiteManager.shouldAddGoogleTopSite(hasSpace: true)
+        return !historySites.isEmpty || (historySites.isEmpty && hasOtherData)
```

Comment:
q. I am wondering if we need to check history data to always be empty for `(historySites.isEmpty && hasOtherData)` case?

If there is no `hasOtherData` and historySites is empty then do we want to return true for hasData? 

```
// result: true
//let historySites = true
//let hasOtherData = true

// result: false
//let historySites = true
//let hasOtherData = false
//
// result: true
//let historySites = false
//let hasOtherData = true
//
// result: true
//let historySites = false
//let hasOtherData = false
```

Reply:
It should be the following:

// result hasData: true
//let historySites = true
//let hasOtherData = true

// result hasData: true
//let historySites = true
//let hasOtherData = false
//
// result hasData: true
//let historySites = false
//let hasOtherData = true
//
// result hasData: false
//let historySites = false
//let hasOtherData = false

Reply:
Can we add unit test to cover all this cases? is simpler than manually testing :)